app/javascript/mastodon/features/emoji/emoji.js

```diff
@@ -73,7 +80,7 @@ const emojify = (str, customEmojis = {}) => {
     } else { // matched to unicode emoji
       const { filename, shortCode } = unicodeMapping[match];
       const title = shortCode ? `:${shortCode}:` : '';
-      replacement = `<img draggable="false" class="emojione" alt="${match}" title="${title}" src="${assetHost}/emoji/${emojiFilename(filename)}.svg" />`;
+      replacement = `<picture><source media="(prefers-color-scheme: dark)" srcset="${assetHost}/emoji/${emojiFilenameInSystemDarkMode(filename)}.svg" /><img draggable="false" class="emojione" alt="${match}" title="${title}" src="${assetHost}/emoji/${emojiFilenameInSystemLightMode(filename)}.svg" /></picture>`;
```

Comment:
Sounds good when the system/automatic setting is selected, but I'm not sure it's the best way to deal with it when a theme is actually selected.

Reply:
Got it. Do you have suggestions for how this could be better dealt with? Or better yet, would love to hear your concerns with this approach.

In my testing, this still behaved as expected for non-system themes—the only downside is that there's a little bit more markup.

Reply:
Resolving per https://github.com/mastodon/mastodon/pull/20107#issuecomment-1318599557.